### PR TITLE
feat(mae/mcl): Make ingestAspect produce both MCLs and MAEs

### PR DESCRIPTION
**Pull request by dexter-mh-lee**
We are migrating away from the legacy MCE/MAE setup to MCP/MCL setup detailed in this doc https://datahubproject.io/docs/advanced/mcp-mcl/

As the first step, we will change ingestAspect to produce MCLs and MAEs, and remove MAE processor to consume only from MCLs. 

We are leaving MAEs as is for now, since there are external dependencies outside of DataHub that may be consuming MAEs. Note, the recommendation is to move to MCLs. 

Tested locally and made sure MCLs are created correctly and that MAEs are still correctly produced. 

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java [hunk 1]
```diff
@@ -208,15 +210,18 @@ public RecordTemplate ingestAspect(@Nonnull final Urn urn, @Nonnull final String
...
+      String entityName = urnToEntityName(urn);
+      EntitySpec entitySpec = getEntityRegistry().getEntitySpec(entityName);
+      AspectSpec aspectSpec = entitySpec.getAspectSpec(aspectName);
+      if (aspectSpec == null) {
+        throw new RuntimeException(String.format("Unknown aspect %s for entity %s", aspectName, entityName));
       }
-      produceMAETimer.stop();
+
+      Timer.Context produceMCLTimer = MetricUtils.timer(this.getClass(), "produceMCL").time();
+      produceMetadataChangeLog(urn, entityName, aspectName, aspectSpec, oldValue, updatedValue, oldSystemMetadata,
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:222]: Can we produce to BOTH? (And deprecate consumption via MAE processor?)

Comment (dexter-mh-lee) [reply metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:222]: Added produce to both and removed MAE consumer

Comment (dexter-mh-lee) [reply metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:222]: Note the produce calls are async under the hood so no impact on latency

File: metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java [hunk 2]
```diff
@@ -388,24 +400,35 @@ public Entity getEntity(@Nonnull final Urn urn, @Nonnull final Set<String> aspec
...
+      if (oldAspectValue != null) {
+        oldSnapshot = buildSnapshot(urn, oldAspectValue);
+      }
+      _producer.produceMetadataAuditEvent(urn, oldSnapshot, newSnapshot, oldSystemMetadata, newSystemMetadata,
+          operation);
     }
+  }
 
-    _producer.produceMetadataAuditEvent(urn, oldSnapshot, newSnapshot, oldSystemMetadata, newSystemMetadata, operation);
+  protected Snapshot buildKeySnapshot(@Nonnull final Urn urn) {
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:421]: Nit: this should prob go below produceMetadataAuditEventForKey

File: metadata-io/src/test/java/com/linkedin/metadata/entity/EbeanEntityServiceTest.java [hunk 3]
```diff
@@ -143,8 +147,17 @@ public void testIngestGetEntity() throws Exception {
     assertTrue(DataTemplateUtil.areEqual(expectedKey,
         readEntity.getValue().getCorpUserSnapshot().getAspects().get(0).getCorpUserKey())); // Key + Info aspect.
 
+    ArgumentCaptor<MetadataChangeLog> mclCaptor = ArgumentCaptor.forClass(MetadataChangeLog.class);
```
Comment (jjoyce0510) [inline metadata-io/src/test/java/com/linkedin/metadata/entity/EbeanEntityServiceTest.java:150]: nice!!

File: metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/MaeConsumerApplication.java [hunk 4]
```diff
@@ -9,7 +9,7 @@
 public class MaeConsumerApplication {
 
   public static void main(String[] args) {
-    Class<?>[] primarySources = {MaeConsumerApplication.class, com.linkedin.metadata.kafka.MaeConsumerConfig.class};
+    Class<?>[] primarySources = {MaeConsumerApplication.class, MclConsumerConfig.class};
```
Comment (jjoyce0510) [inline metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/MaeConsumerApplication.java:12]: Yess!

**Review by jjoyce0510 — APPROVED** [review]
LGTM

**Review by shirshanka — APPROVED** [review]

**Pull request merged**